docker-bake.hcl

```diff
@@ -14,7 +14,7 @@ variable "DOCKER_BUILDTAGS" {
   default = ""
 }
 variable "DOCKER_GITCOMMIT" {
-  default = null
+  default = ""
```

Comment:
We need to keep `null` here otherwise would set empty string instead of defaulting to HEAD in the Dockerfile.

Reply:
Hmm; the previous version would have defaulted to `GITHUB_SHA` due to `DOCKER_GITCOMMIT != null ? DOCKER_GITCOMMIT : GITHUB_SHA`... Do we think the `HEAD` fallback is better than https://github.com/moby/moby/pull/46267/commits/d076752827acdb2e5a4920c94365f40a35a05c0f?

Reply:
I agree that `HEAD` is not very informative, (head of ... what?)

Reply:
> Also, is that functionality even useful now? Most methods of building the daemon will not be able to detect a dirty tree, as the Git repo is not in the build context.

If you use our scripts, perhaps not, but I for  sure have seen quite some "not-built-by-us" packages that were built from dirty state (which is why we added the `-unsupported` for)

I think a `dev` as version (or `0.0.0-dev`) and `<unknown>` (or something) as commit would be ok;

- it's a "dev" build if no version was specified
- and we don't know what the commit is that was built from if it's not specified (or no `.git` around)

Reply:
Sorry for the confusion what I meant is for local builds we are still going to have this error https://github.com/moby/moby/actions/runs/5903693397/job/16014198814#step:4:783 if `DOCKER_GITCOMMIT` is not set (`docker buildx bake`).

```
#23 [build 6/6] RUN --mount=type=bind,target=.,rw     --mount=type=tmpfs,target=cli/winresources/dockerd     --mount=type=tmpfs,target=cli/winresources/docker-proxy     --mount=type=cache,target=/root/.cache/go-build,id=moby-build-linux/amd64 <<EOT (set -e...)
#23 0.474 
#23 0.479 error: .git directory missing and DOCKER_GITCOMMIT not specified
#23 0.479   Please either build with the .git directory accessible, or specify the
#23 0.479   exact (--short) commit hash you are building using DOCKER_GITCOMMIT for
#23 0.479   future accountability in diagnosing build issues.  Thanks!
```